doozer/doozerlib/cli/scan_sources_konflux.py

```diff
@@ -905,6 +910,196 @@ async def scan_extra_packages(self, image_meta: ImageMetadata):
+                    #   - At 15 days: 1 in 16 chance (~6%)
+                    #   - At 20 days: 1 in 11 chance (~9%)
+                    #   - At 25 days: 1 in 6 chance (~17%)
+                    #   - At 29 days: 1 in 2 chance (50%)
+                    #   - At 30+ days: Always rebuild (100%)
+                    self.logger.info(
+                        f'Task bundle {task_name} is {task_age_days} days old (>= 10 days), applying staggered rebuild logic'
+                    )
+                    rebuild_probability_denominator = max(30 - task_age_days, 1)
+                    should_rebuild = random.randint(1, rebuild_probability_denominator) == 1
+
+                    if should_rebuild:
+                        self.logger.info(
+                            f'Triggering rebuild for {image_meta.distgit_key} due to outdated task bundle {task_name} ({task_age_days} days old)'
+                        )
+                        self.add_image_meta_change(
+                            image_meta,
+                            RebuildHint(
+                                RebuildHintCode.TASK_BUNDLE_OUTDATED,
+                                f'Task bundle {task_name} is {task_age_days} days old (>={10} days) '
+                                f'and newer version is available (staggered rebuild)',
+                            ),
+                        )
+                        return
+                    else:
+                        self.logger.info(
+                            f'Task bundle {task_name} is {task_age_days} days old but staggered rebuild '
+                            f'logic decided not to rebuild (probability was 1/{rebuild_probability_denominator})'
+                        )
+                elif task_age_days:
+                    self.logger.info(
+                        f'Task bundle {task_name} is only {task_age_days} days old (< 10 days), skipping rebuild'
+                    )
+            else:
+                self.logger.info(f'Task bundle {task_name} is up to date (SHA: {used_sha[:12]}...)')
+
+    async def get_current_task_bundle_shas(self) -> Dict[str, str]:
+        """
+        Fetch current task bundle SHAs from the art-konflux-template GitHub repository
+        """
+        self.logger.info(f'Fetching task bundle template from {KONFLUX_DEFAULT_IMAGE_BUILD_PLR_TEMPLATE_URL}')
+        try:
+            async with self.session.get(
+                KONFLUX_DEFAULT_IMAGE_BUILD_PLR_TEMPLATE_URL, headers={'Authorization': f'Bearer {self.github_token}'}
+            ) as response:
+                response.raise_for_status()
+                yaml_content = await response.text()
+
+            # Parse YAML to extract task bundle references
+            self.logger.info('Parsing YAML content to extract task bundle references')
+            yaml_data = yaml.safe_load(yaml_content)
+            task_bundles = {}
+
+            # Look for task references in the YAML
+            def extract_task_refs(obj):
```

Comment:
a general "return early" style suggestion in this func to make it more readable:

```
def func():
  if condition_a:
    for loop:
      if condition_b:
        do_something

def func():
  if not condition_a:
    return
  for loop:
    if not condition_b:
      continue
    do_something
```

Reply:
makes sense - added in https://github.com/openshift-eng/art-tools/pull/1877/commits/f7634f5543f1448b08c64884da7e34da28cc64ae